src/enclave/App/CMakeLists.txt

```diff
@@ -7,28 +7,23 @@ set(SOURCES
   ${CMAKE_CURRENT_BINARY_DIR}/Enclave_u.c)
 
 add_custom_command(
-  COMMAND ${SGX_EDGER8R} --untrusted ${CMAKE_SOURCE_DIR}/Enclave/Enclave.edl --search-path ${CMAKE_SOURCE_DIR}/Enclave --search-path "$ENV{SGX_SDK}/include"
+  COMMAND oeedger8r --untrusted ${CMAKE_SOURCE_DIR}/Enclave/Enclave.edl --search-path ${CMAKE_SOURCE_DIR}/Enclave
   DEPENDS ${CMAKE_SOURCE_DIR}/Enclave/Enclave.edl
-  OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/Enclave_u.c)
+  OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/Enclave_u.h ${CMAKE_CURRENT_BINARY_DIR}/Enclave_u.c ${CMAKE_CURRENT_BINARY_DIR}/Enclave_args.h)
 
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC -Wno-attributes")
 set(CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS} ${CMAKE_CXX_FLAGS}")
 
 add_library(enclave_jni SHARED ${SOURCES})
 
-find_library(UKEY_EXCHANGE_LIB sgx_ukey_exchange)
-find_library(URTS_LIB sgx_urts)
-find_library(URTS_SIM_LIB sgx_urts_sim)
-find_library(UAE_SERVICE_LIB sgx_uae_service)
-find_library(UAE_SERVICE_SIM_LIB sgx_uae_service_sim)
-
-target_link_libraries(enclave_jni ${UKEY_EXCHANGE_LIB} pthread)
-if(NOT "$ENV{SGX_MODE}" STREQUAL "HW")
-  message(STATUS "Building for simulated SGX")
-  target_link_libraries(enclave_jni ${URTS_SIM_LIB} ${UAE_SERVICE_SIM_LIB})
-else()
-  message(STATUS "Building for SGX hardware")
-  target_link_libraries(enclave_jni ${URTS_LIB} ${UAE_SERVICE_LIB})
+target_include_directories(enclave_jni PRIVATE
+# Needed for the generated file helloworld_u.h
+  ${CMAKE_CURRENT_BINARY_DIR})
+
+if ("$ENV{MODE}" STREQUAL "SIMULATE")
+  target_compile_definitions(enclave_jni PUBLIC -DSIMULATE)
```

Comment:
1. Do we need to do anything for hardware mode here?
2. It would be nice to produce an error if `$ENV{MODE}` is anything other than the two accepted values.

Reply:
By default, the program compiles in hardware mode, unless the MODE is set to "SIMULATE", in which case the code will run in simulation mode.